src/tojson.jl

```diff
@@ -84,6 +84,23 @@ function tojson(output::String, ts::ReportingTestSet)
     # All stdout from the top level test set, used for all tests.
     output = truncate_output(output)
 
+    function test_code(result::Test.Result)
+        if result isa Test.Pass && result.test_type == :test_throws
+            "@test_throws $(result.data) $(result.orig_expr)"
+        elseif result isa Test.Fail && startswith(string(result.test_type), "test_throws")
+            "@test_throws $(result.data) $(result.orig_expr)"
+        elseif result isa Test.LogTestFailure
+            "@test_logs $(join(result.patterns, ' ')) $(result.orig_expr)"
+        elseif result isa Test.Error && result.test_type == :test_unbroken
+            "@test_broken $(result.orig_expr)"
+        elseif result isa Test.Broken
+            macro_name = result.test_type === :skipped ? "@test_skip " : "@test_broken "
+            "$macro_name $(result.orig_expr)"
+        else
+            "@test $(result.orig_expr)"
+        end
+    end
+
```

Comment:
To be honest, I find this a bit hard to read due to the long conditions.

What do you think about splitting this up into several methods dispatching based on the type of result instead? That would simplify it a lot, I think.

Reply:
I'd slightly prefer that suggestion.